functions/openhab.js

```diff
@@ -430,14 +511,16 @@ function syncAndDiscoverDevices(token, success, failure) {
 
 		// Now retrieve all other items
 		(function () {
+			itemLoop:
 			for (var itemNum in items) {
```

Comment:
Be careful with `for ... in` loops.
It definitely should not iterate over Array elements. 
See [this resource](https://developer.mozilla.org/pl/docs/Web/JavaScript/Reference/Statements/for...in).

I would suggest using `forEach` if `items` is an Array:

```
items.forEach(function(item) {
});
```

If it's not, we should check if `items.hasOwnProperty(itemNum)` and then proceed with the rest.